handler/src/main/java/io/netty/handler/pcap/PcapWriteHandler.java

```diff
@@ -0,0 +1,523 @@
+                return;
+            }
+
+            // Write Packet into Pcap
+            pCapWriter.writePacket(pcap, ethernetBuf);
+        } catch (IOException ex) {
+            logger.error("Caught Exception While Writing Packet into Pcap", ex);
+            ctx.fireExceptionCaught(ex);
+        } finally {
+            ipBuf.release();
+            ethernetBuf.release();
+            pcap.release();
+        }
+    }
+
+    @Override
+    public void handlerRemoved(ChannelHandlerContext ctx) throws Exception {
+
+        // If `isTCP` is true, then we'll simulate a `FIN` flow.
+        if (ctx.channel() instanceof SocketChannel) {
+            logger.debug("Starting Fake TCP FIN+ACK Flow to close connection");
+
+            ByteBufAllocator byteBufAllocator = ctx.alloc();
+            ByteBuf tcpBuf = byteBufAllocator.buffer();
+
+            try {
+                // Write FIN+ACK with Normal Source and Destination Address
+                TCPPacket.writePacket(tcpBuf, null, sendSegmentNumber, receiveSegmentNumber, srcAddr.getPort(),
+                        dstAddr.getPort(), TCPPacket.TCPFlag.FIN, TCPPacket.TCPFlag.ACK);
+                completeTCPWrite(srcAddr, dstAddr, tcpBuf, byteBufAllocator, ctx);
+
+                // Write FIN+ACK with Reversed Source and Destination Address
+                TCPPacket.writePacket(tcpBuf, null, receiveSegmentNumber, sendSegmentNumber, dstAddr.getPort(),
+                        srcAddr.getPort(), TCPPacket.TCPFlag.FIN, TCPPacket.TCPFlag.ACK);
+                completeTCPWrite(dstAddr, srcAddr, tcpBuf, byteBufAllocator, ctx);
+
+                // Write ACK with Normal Source and Destination Address
+                TCPPacket.writePacket(tcpBuf, null, sendSegmentNumber + 1, receiveSegmentNumber + 1,
+                        srcAddr.getPort(), dstAddr.getPort(), TCPPacket.TCPFlag.ACK);
+                completeTCPWrite(srcAddr, dstAddr, tcpBuf, byteBufAllocator, ctx);
+            } finally {
+                tcpBuf.release();
+            }
+
+            logger.debug("Finished Fake TCP FIN+ACK Flow to close connection");
+        }
+
+        this.pCapWriter.close();
+        super.handlerRemoved(ctx);
+    }
+
+    @Override
+    public void exceptionCaught(ChannelHandlerContext ctx, Throwable cause) throws Exception {
+
+        if (ctx.channel() instanceof SocketChannel) {
+            ByteBuf tcpBuf = ctx.alloc().buffer();
+
+            try {
+                // Write RST with Normal Source and Destination Address
+                TCPPacket.writePacket(tcpBuf, null, sendSegmentNumber, receiveSegmentNumber, srcAddr.getPort(),
+                        dstAddr.getPort(), TCPPacket.TCPFlag.RST, TCPPacket.TCPFlag.ACK);
+                completeTCPWrite(srcAddr, dstAddr, tcpBuf, ctx.alloc(), ctx);
+            } finally {
+                tcpBuf.release();
+            }
+
+            logger.debug("Sent Fake TCP RST to close connection");
+        }
+
+        this.pCapWriter.close();
```

Comment:
@normanmaurer Do I need to catch this `IOException` and fire 1 more `ctx.fireExceptionCaught(ex)`?